lib/screens/scanner_screen.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:qrdemo/providers/firestore_provider.dart';
+import 'package:qrdemo/providers/routing_provider.dart';
+
+class ScannerScreen extends StatefulWidget {
+  ScannerScreen({Key? key}) : super(key: key);
+
+  @override
+  _ScannerScreenState createState() => _ScannerScreenState();
+}
+
+class _ScannerScreenState extends State<ScannerScreen> {
+  final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
+
+  QRViewController? controller;
+  Barcode? result;
+
+  @override
+  Widget build(BuildContext context) {
+    return QRView(
+      key: qrKey,
+      onQRViewCreated: _onQRViewCreated,
+    );
+  }
+
+  void _onQRViewCreated(QRViewController controller) {
+    this.controller = controller;
+    controller.scannedDataStream.listen((scanData) async {
+      await controller.pauseCamera();
+      Provider.of<FirestoreProvider>(context, listen: false)
+          .confirmScan(scanData, context);
+      Provider.of<RoutingProvider>(context, listen: false)
+          .goToHomeScreen(context);
```

Comment:
the snackbar gets its context from the QRView created when the user presses scan. We pass the context there in confirmScan() and the snackbar creates itself using the Build Context it is provider. The snackbar will show once the listener detects a scan, and depending on the string, it opens a certain text. I'm assuming that when it is navigating away the context is still active because QRView is still in the stack of routes. I also read that the Flutter team has been working on snackbars and making it more efficient. Not entirely sure, but I assume that snackbars are visible on ALL screens until they are dismissed.

Reply:
Good catch, though, because I never thought about it like that, and I can see how this could be seen as warning.